### PR TITLE
added a sub workflow to solve issue #121 in order to obtain the gene statistics from an annotation file (gff3)

### PR DESCRIPTION
**Pull request by SandraBabirye**
1. Added a new parameter --annotation_set  in the config files 
2. Used nf-core modules (agat-spstatistics and agat-sqstatbasic" to obtain the statistics of the features for example: gene, exons, introns, CDS , mRNA
3. Added a new sub workflow (annotation_stats.nf)

### CODE REVIEW

File: conf/base.config
Comment (BethYates) [inline conf/base.config:1]: Theres a typo withName: "AGT_SPSTATISTICS" should be "withName: AGAT_SPSTATISTIC"

File: workflows/genomenote.nf
Comment (BethYates) [inline workflows/genomenote.nf:1]: Lines 69 and 70 are not needed as you don't use these modules in this workflow, they are only called in the ANNOTATION_STATS subworkflow

Comment (BethYates) [inline workflows/genomenote.nf:1]: There is a typo on line 142 you are missing the closing parenthesis ")"

File: nextflow.config
Comment (BethYates) [inline nextflow.config:1]: When you add a new parameter you also need to nextflow.config you also need to add a description of it to nextflow_schema.json

File: subworkflows/local/annotation_statistics.nf
Comment (BethYates) [inline subworkflows/local/annotation_statistics.nf:1]: As you are calling this subworkflow from the genomenote.nf workflow you are passing it the ch_gff3 channel as input - you create this channel in genomenote.nf so lines 10 and 11 are not needed. 

Instead modify the workflow block to include "take:", "main:" and "emit:" blocks, try looking at the GENOME_STATISTICS.nf subworkflow to see how this works. 

You should also be outputting the version information for each of the tools this subworkflow is using - see section 1.2 here https://nf-co.re/docs/guidelines/components/subworkflows

File: workflows/genomenote.nf
Comment (SandraBabirye) [reply workflows/genomenote.nf:1]: Thank you I have resolve this added the clossing parenthesis

**Comment by BethYates** [comment]
Superseded by #133

**Pull request closed without merging**